### DISCUSSION TITLE
How to define different MinHeight in ControlTheme

### DISCUSSION THREAD

**Discussion by Bruce9009** [AvaloniaUI/Avalonia#16155] [Q&A] [answered]
I would like to define a custom Button based on Avalonia.Controls.Button, so firstly I inherited it without registering any new properties yet.
![image](https://github.com/AvaloniaUI/Avalonia/assets/101933314/2e5bc048-a612-442d-9f9e-8b6e37f3837b)

Then I created a ResourceDictionary file to define a ControlTheme for my Button. The intention is simple...I just want to define different sizes by using MinHeight property, so the codes look like as follows:
![image](https://github.com/AvaloniaUI/Avalonia/assets/101933314/87f192b7-c799-4d24-b375-4967b3257977)

But when I tried to call the defined styles and had a preview, I found that it did not work. All buttons have a same MinHeight...And it seems that the value of MinHeight was set to the Large one which was also the last one defined...
![image](https://github.com/AvaloniaUI/Avalonia/assets/101933314/cf0cb454-ca09-4d7b-b3b8-f118f98ae811)

![image](https://github.com/AvaloniaUI/Avalonia/assets/101933314/03794da0-d79c-46f5-8dae-6329b7d51473)

I think it is a very simple use case and I must miss something...
Does anyone have any ideas about how to solve this question? Thanks in advance!

**Comment by stevemonaco** [comment, marked as answer]
The default `VerticalAlignment` is `Stretch`. Which means your custom buttons are being stretched as much as the parent container allows. `MinHeight` only constrains the button from being too small.

See https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Avalonia.Themes.Fluent/Controls/Button.xaml#L21 for setters on `VerticalAlignment` and `HorizontalAlignment`.

**Reply by Bruce9009** [reply]
Yes, you are right. After setting `VerticalAlignment` as `Center`, it works as expected!
Thank you very much!
